src/vis_cpu/vis_cpu.py

```diff
@@ -227,6 +227,12 @@ def vis_cpu(
             "beam type must be power and have only one pol (either xx or yy) if polarized=False"
         )
 
+    # Need to make sure that the beams are peak normalized for output to make sense.
```

Comment:
Why is this? If a beam is less that unity everywhere, doesn't that make sense?

Also, I feel like I remember we tried to not change the input beams in-place (so eg. when we call `efield_to_power()` we do it on copies of the beams). Not sure how important this is, but we should think about it.

Reply:
I think the reasoning is that in the visibility equation, the antenna response is supposed to be peak-normalized. At the very least, this is what they do in `pyuvsim`. If we wanted to not change it in place, then I suppose we could throw in something like `beam_list = [copy.deepcopy(beam) for beam in beam_list]`, though that increases the memory needs.

Reply:
I'm... not sure that this is required? Doesn't it depend on how you treat the bandpass? Can maybe @bhazelton chime in here?

Reply:
I'm pretty sure this happens for every `pyuvsim` simulation. When the `pyuvsim.Antenna` class calculates the Jones matrix, it makes sure that the associated beam is peak-normalized. See this [code snippet](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/main/pyuvsim/antenna.py#L75).

Reply:
I think we have an open issue to change that. It's not necessarily the right thing to be doing.

Reply:
(depends on what the user wants, but peak normalizing removes all info from the sim about the bandpass, which can be really useful info).

Reply:
Maybe we could issue a warning and let the user know the bandpass can be accessed in the peak-normalized beam's `bandpass_array` attribute? Or write the bandpass somewhere? One thing I know for sure is that the visibilities coming out of `vis_cpu` for the current validation effort look like they're too bright, and a quick comparison against some plots of the data I made for the last validation effort suggests that they're off by about a factor of the per-antenna beam squared.

Reply:
I think maybe this should be an option at the hera_sim level? I.e. have a parameter `peak_normalize_beam = False` in the `VisibilitySimulator` class. I think probably vis_cpu should just do what it's told. Since the default is False, setting to True means the user is aware that they're changing the beam data internally.

Reply:
Hm. I think I could get behind that. Do you think we should bother with issuing a warning in the case of non-peak-normalized beams, or just assume that the user knows what they're doing and has their reasons?

Reply:
OK, no warning or anything will be issued. We can revisit this later if it ends up being an issue.